Formula/cadence-workflow.rb

```diff
@@ -0,0 +1,33 @@
+class CadenceWorkflow < Formula
+  desc "Cadence Workflow is a distributed, scalable, durable, and highly available orchestration engine to execute asynchronous long-running business logic in a scalable and resilient way"
+  homepage "https://cadenceworkflow.io/"
+  url "https://github.com/uber/cadence.git",
+      tag:      "v0.18.1",
+      revision: "b6dfe37ebfec7742b1470be7129f07518e8bec98"
+  license "MIT"
+
+  bottle do
+  end
+
+  depends_on "go" => :build
+
+  conflicts_with "cadence", because: "both install an `cadence` executable"
+
+  def install
+    ENV["PATH"] = "#{ENV["PATH"]}:#{ENV["GOPATH"]}/bin"
+    system "make", "cadence", "cadence-server", "cadence-canary", "cadence-sql-tool", "cadence-cassandra-tool"
+    bin.install "cadence"
+    bin.install "cadence-server"
+    bin.install "cadence-canary"
+    bin.install "cadence-sql-tool"
+    bin.install "cadence-cassandra-tool"
+
+    system "cp", "-r", "config", "#{etc}/cadence"
+    system "cp", "-r", "schema", "#{etc}/cadence"
+  end
+
+  test do
+    assert_match "cadence:",
+      shell_output("#{bin}/cadence version")
```

Comment:
Can you propose a more substantial test than checking the version or calling for help? These tests rarely ever fail even for broken installations.

>We want tests that don’t require any user input and test the basic functionality of the application. For example `foo build-foo input.foo` is a good test and (despite their widespread use) `foo --version` and `foo --help` are bad tests. 

Ref. https://docs.brew.sh/Formula-Cookbook#add-a-test-to-the-formula

Reply:
Added:
```
(cadence)$brew test cadence-workflow
==> Testing cadence-workflow
==> /usr/local/Cellar/cadence-workflow/0.18.2/bin/cadence --version
```

(btw this is a little werid that the project release version is 0.18.2 while the assertion version of `cadence` is 0.18.3
That's because we are missing the version tag in server command right now which will be fixed soon by https://github.com/uber/cadence/issues/4052
I will update this test later after that fix is released.